Use netscape.security from org.mozilla.jss

cipherboy opened the pull request:
Note that this PR is pending JSS 4.5.2 patchset 2 being released in Fedora. The spec file will also need to be bumped. 

This updates all imports to use the `org.mozilla.jss` version of the
`netscape.security.*` packages. This is to remove code duplicated between
PKI and JSS, finishing the work from jss commit
485a5b6713c9fb087804b687da895e33775b2f4b.

The only non-obvious fixup is in `*.def.CertificatePoliciesExtDefault`,
where a local variable "org" was preventing `org.mozilla.jss.*` from
being used.

Except for the one non-obvious fix up, the rest was performed automatically. 

`Signed-off-by: Alexander Scheel <ascheel@redhat.com>`

cipherboy commented:
TODO: Handle `com.netscape.cmsutil.util.{Cert,Utils}`

edewata commented on base/ca/shared/conf/CS.cfg:
```diff
@@ -957,23 +957,23 @@ log.instance.Transactions.rolloverInterval=2592000
 logAudit.fileName=[PKI_INSTANCE_PATH]/logs/[PKI_SUBSYSTEM_TYPE]/access
 logError.fileName=[PKI_INSTANCE_PATH]/logs/[PKI_SUBSYSTEM_TYPE]/error
-oidmap.auth_info_access.class=netscape.security.extensions.AuthInfoAccessExtension
+oidmap.auth_info_access.class=org.mozilla.jss.netscape.security.extensions.AuthInfoAccessExtension
```
Assuming the old classes will no longer exist, we need to provide an upgrade script so that existing instances will continue to work.

cipherboy replied:
Ahh right, these are configurations.

#122 removes the old classes, so yes, I'll try creating an upgrade script.

cipherboy commented:
@edewata Like this? How would I go about testing the migration?

edewata commented:
@cipherboy Yes, it looks good.

If you want to test a real upgrade, install an older version (e.g. PKI 10.6.9), create an instance, inspect the configuration, install the new version (i.e. PKI 10.7.0), then verify the configuration changes.

If you just want to do a quicker test, install PKI 10.7.0, change the properties to look like PKI 10.6.9, reinstall PKI 10.7.0, then verify the configuration changes.

cipherboy commented:
Rebased off of master, added `import org.mozilla.jss.netscape.security.util.Utils;` to `Keyring.java`.

cipherboy commented:
@edewata Ok, this PR is ready for final review. I realized that in my rebase I dropped the upgrade script. 

I realized I had spare commas at the end of some of the properties we wanted to update. I fixed that as well, so the upgrade works correctly now. 


Post-upgrade:
```
[root@localhost pki-tomcat]# grep -ir '=org.mozilla.jss'
ca/CS.cfg:oidmap.auth_info_access.class=org.mozilla.jss.netscape.security.extensions.AuthInfoAccessExtension
ca/CS.cfg:oidmap.extended_key_usage.class=org.mozilla.jss.netscape.security.extensions.ExtendedKeyUsageExtension
ca/CS.cfg:oidmap.netscape_comment.class=org.mozilla.jss.netscape.security.x509.NSCCommentExtension
ca/CS.cfg:oidmap.ocsp_no_check.class=org.mozilla.jss.netscape.security.extensions.OCSPNoCheckExtension
ca/CS.cfg:oidmap.pse.class=org.mozilla.jss.netscape.security.extensions.PresenceServerExtension
ca/CS.cfg:oidmap.subject_info_access.class=org.mozilla.jss.netscape.security.extensions.SubjectInfoAccessExtension
```

```
[root@localhost pki-tomcat]# grep -ir '=netscape.security'
ca/archives/CS.cfg.bak.20190222151828:oidmap.auth_info_access.class=netscape.security.extensions.AuthInfoAccessExtension
ca/archives/CS.cfg.bak.20190222151828:oidmap.extended_key_usage.class=netscape.security.extensions.ExtendedKeyUsageExtension
ca/archives/CS.cfg.bak.20190222151828:oidmap.netscape_comment.class=netscape.security.x509.NSCCommentExtension
ca/archives/CS.cfg.bak.20190222151828:oidmap.ocsp_no_check.class=netscape.security.extensions.OCSPNoCheckExtension
ca/archives/CS.cfg.bak.20190222151828:oidmap.pse.class=netscape.security.extensions.PresenceServerExtension
ca/archives/CS.cfg.bak.20190222151828:oidmap.subject_info_access.class=netscape.security.extensions.SubjectInfoAccessExtension
```

jmagne commented:
I just scrolled through this whole thing. Looks good from here. Took a closer look at the upgrade script and looked good.
